runtime/src/bank.rs

```diff
@@ -1826,19 +1826,38 @@ impl Bank {
 
     /// Recalculate the hash_internal_state from the account stores. Would be used to verify a
     /// snapshot.
-    pub fn verify_hash_internal_state(&self) -> bool {
+    #[must_use]
+    fn verify_bank_hash(&self) -> bool {
         self.rc
             .accounts
             .verify_bank_hash(self.slot(), &self.ancestors)
     }
 
+    #[must_use]
+    fn verify_hash(&self) -> bool {
+        assert!(self.is_frozen());
+        let calculated_hash = self.hash_internal_state();
```

Comment:
Just for the record: I'm little concerned about the verification failure for the restarted (hard-forked) cluster, but it should not be a problem because the renewed hard_fork information itself should be included in the bank of updated snapshot. Current ops doesn't use `solana-validator`'s `--hard-fork`.

Reply:
`solana-validator`'s `--hard-fork` is only there for a validator that never wants to consume a snapshot, for example a node that wants to archive the entire ledger to long-term storage for posterity.

Reply:
Oh, I see. Then, necessitated manual restarts for the hard-fork should be acceptable for such a scenario. Thanks for clarification!